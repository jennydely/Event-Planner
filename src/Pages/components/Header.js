import React, { useContext } from 'react'
import styled from 'styled-components/macro'
import SelectEvents from './common/SelectEvents'
import EventListFilter from './common/EventListFilter'
import CategoryFilter from './CategoryFilter'
import Searchbar from './Searchbar'
import { UserContext } from "../../providers/UserProvider";

export default function Header({
  onSelectFilter,
  onSelectEventFilter,
  onSelectEventListFilter,
  eventArray,
  eventFilter,
  chosenEventListFilter,
  categoryFilter,
  handleEventSuggestion,
  handleEventSearch,
}) {
  const hasHiddenEvent = eventArray.some((event) => event.isHidden === true)
  const hasOldEvent = eventArray.some(
    (event) => event.eventEndDate < new Date().toJSON().slice(0, 10)
  )
  const sortOptions = ['Sort by', 'Date', 'A-z', 'Z-a']
  if (hasHiddenEvent) sortOptions.push('Hidden')
  if (hasOldEvent) sortOptions.push('Old')

  const chosenEventList = [{ name: 'Public events', value: 'Public' }, { name: 'Favorite events', value: 'Favorite' }, { name: 'Private events', value: 'Private' }]
  const user = useContext(UserContext);
  console.log('headerUser', user)
  return (
    <StyledHeader>
      <FilterContainer>
        <SelectEvents
          id="filter"
          name={sortOptions}
          options={sortOptions}
          onSelectEventFilter={onSelectEventFilter}
          value={eventFilter}
        />
        <Searchbar
          handleEventSuggestion={handleEventSuggestion}
          handleEventSearch={handleEventSearch}
        />
        <CategoryFilter
          onSelectFilter={onSelectFilter}
          categoryFilter={categoryFilter}
        />
      </FilterContainer>
      {   user ? <EventListFilter
        onSelectEventListFilter={onSelectEventListFilter}
        id="EventListfilter"
        name={chosenEventList}
        options={chosenEventList}
        value={chosenEventListFilter} /> : ''}

    </StyledHeader>
  )
}
const StyledHeader = styled.header`
`


const FilterContainer = styled.div`
   max-height: 55px;
   display: flex;
    flex-direction: row;
    align-items:flex-end;
    width: 370px;
    justify-content:space-between;
`