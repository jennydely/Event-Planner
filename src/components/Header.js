import React from 'react'
import styled from 'styled-components/macro'
import CategoryFilter from './CategoryFilter'
import SelectEvents from '../common/SelectEvents'

export default function Header({
  onSelectFilter,
  onSelectEventFilter,
  hasHiddenEvent,
  hasOldEvent,
}) {
  const sortOptions = ['date', 'a-Z', 'Z-a']
  if (hasHiddenEvent) sortOptions.push('Hidden')
  if (hasOldEvent) sortOptions.push('Old')

  return (
    <header>
      <ButtonContainer>
        <SelectEvents
          id="filter"
          name={sortOptions}
          options={sortOptions}
          onSelectEventFilter={onSelectEventFilter}
        />
      </ButtonContainer>
      <CategoryFilter onSelectFilter={onSelectFilter} />
    </header>
  )
}

const ButtonContainer = styled.div`
  margin: 5px;
`
