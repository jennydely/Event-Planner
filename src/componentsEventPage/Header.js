import React from 'react';
import styled from 'styled-components/macro'
import CategoryFilter from './CategoryFilter'

export default function Header({ onSelectFilter }) {
    return (
        <header>
            <ButtonContainer>
                <button name="all" onClick={() => onSelectFilter('all')}>All</button>
            </ButtonContainer>
            <CategoryFilter onSelectFilter={onSelectFilter} />
        </header>
    )
}

const ButtonContainer = styled.div`
   margin: 5px;
   width: wh%;
`