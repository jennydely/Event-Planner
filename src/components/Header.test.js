import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'
import 'jest-styled-components'

describe('Header', () => {
  const onSelectFilter = 'all'

  it('displays the sort by date function', () => {
    const { getByText } = render(<Header onSelectFilter={onSelectFilter} />)
    expect(getByText('date')).toBeInTheDocument()
  })
  it('renders correctly', () => {
    const tree = render(<Header onSelectFilter={onSelectFilter} />)
    expect(tree).toMatchSnapshot()
  })
})
