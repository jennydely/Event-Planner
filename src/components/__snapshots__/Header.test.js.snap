// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c5 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 100%;
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c6 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 100%;
  color: var(--lightyellow-90);
  background: var(--darkgrey-75);
}

.c7 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 100%;
  color: var(--lightyellow-90);
  background: var(--darkbrown-70);
}

.c8 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 100%;
  color: var(--lightyellow-90);
  background: var(--blue-70);
}

.c9 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  font-size: 100%;
  color: var(--lightyellow-90);
  background: var(--lightbrown-70);
}

.c3 {
  margin: 5px;
}

.c1 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c2 {
  display: block;
  border-radius: 4px;
  border: var(--border-darkgrey);
  color: var(--lightyellow-90);
  background: var(--lightgrey-main);
}

.c0 {
  margin: 5px;
}

<body>
    <div>
      <header>
        <div
          class="c0"
        >
          <select
            class="c1"
            id="filter"
            name="date,a-Z,Z-a"
          >
            <option
              class="c2"
            >
              date
            </option>
            <option
              class="c2"
            >
              a-Z
            </option>
            <option
              class="c2"
            >
              Z-a
            </option>
          </select>
        </div>
        <div
          class="c3"
        >
          <select
            class="c4"
            id="category"
            name="all,metal,medieval,holiday,other"
          >
            <option
              class="c5"
            >
              all
            </option>
            <option
              class="c6"
            >
              metal
            </option>
            <option
              class="c7"
            >
              medieval
            </option>
            <option
              class="c8"
            >
              holiday
            </option>
            <option
              class="c9"
            >
              other
            </option>
          </select>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="Header__ButtonContainer-mnmp8e-0 hThdJr"
      >
        <select
          class="SelectEvents__SelectStyled-sc-11glujg-0 iWWXzJ"
          id="filter"
          name="date,a-Z,Z-a"
        >
          <option
            class="SelectEvents__OptionStyled-sc-11glujg-1 jxhKyn"
          >
            date
          </option>
          <option
            class="SelectEvents__OptionStyled-sc-11glujg-1 jxhKyn"
          >
            a-Z
          </option>
          <option
            class="SelectEvents__OptionStyled-sc-11glujg-1 jxhKyn"
          >
            Z-a
          </option>
        </select>
      </div>
      <div
        class="CategoryFilter__FilterContainer-rogkyz-0 fmiTmv"
      >
        <select
          class="SelectCategory__SelectStyled-sc-1n16935-0 kPEDeF"
          id="category"
          name="all,metal,medieval,holiday,other"
        >
          <option
            class="SelectCategory__OptionStyled-sc-1n16935-1 fMxUKl"
          >
            all
          </option>
          <option
            class="SelectCategory__OptionStyled-sc-1n16935-1 lgliYi"
          >
            metal
          </option>
          <option
            class="SelectCategory__OptionStyled-sc-1n16935-1 gukUCk"
          >
            medieval
          </option>
          <option
            class="SelectCategory__OptionStyled-sc-1n16935-1 cyepkg"
          >
            holiday
          </option>
          <option
            class="SelectCategory__OptionStyled-sc-1n16935-1 hvLasW"
          >
            other
          </option>
        </select>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
